library/Vanilla/DateFilterSchema.php

```diff
@@ -58,54 +89,53 @@ private function parseRange($dates, $open, $close, ValidationField $field) {
         // This notation only allows two dates, specifically.
         $dateArray = explode(',', $dates);
         if (count($dateArray) != 2) {
-            $field->addError('invalid', ['messageCode' => '{field} must be two datetime values.']);
+            $field->addError('invalid', ['messageCode' => '{field} date filter range must contain 2 datetime values.']);
             return Invalid::value();
         }
-        array_walk($dateArray, 'trim');
 
-        if (!preg_match("/{$this->dateRegEx}/i", $dateArray[0], $beginParts) ||
-            !preg_match("/{$this->dateRegEx}/i", $dateArray[1], $endParts)
-        ) {
+        $fakeField = new ValidationField($this->createValidation(), null, null);
+        if (Invalid::isInvalid($this->validateDatetime($dateArray[0], $fakeField))
```

Comment:
`Schema::validateDatetime` will return an "invalid" value or an instance of `DateTimeImmutable`. I believe you could use these return values, instead of attempting to instantiate new instances of `DateTimeImmutable` based on the same input, further down.